nca/Parsers/CalicoPolicyYamlParser.py

```diff
@@ -453,33 +482,65 @@ def _parse_xgress_rule(self, rule, is_ingress, policy_selected_eps, is_profile):
             src_res_pods &= policy_selected_eps
 
         connections = ConnectionSet()
+        conn_props = ConnectivityProperties.make_empty_props()
         if protocol is not None:
+            protocols = ProtocolSet()
+            protocols.add_protocol(protocol)
             if not_protocol is not None:
                 if protocol == not_protocol:
                     self.warning('Protocol and notProtocol are conflicting, no traffic will be matched', rule)
                 else:
                     self.warning('notProtocol field has no effect', rule)
             else:
                 if protocol_supports_ports:
-                    connections.add_connections(protocol, TcpLikeProperties(src_res_ports, dst_res_ports))
+                    connections.add_connections(protocol, ConnectivityProperties.make_conn_props(
+                        self.peer_container, src_ports=src_res_ports, dst_ports=dst_res_ports))
+                    if self.optimized_run != 'false' and src_res_pods and dst_res_pods:
+                        src_num_port_set = PortSet()
+                        src_num_port_set.port_set = src_res_ports.port_set.copy()
+                        dst_num_port_set = PortSet()
+                        dst_num_port_set.port_set = dst_res_ports.port_set.copy()
+                        conn_props = ConnectivityProperties.make_conn_props(self.peer_container,
+                                                                            src_ports=src_num_port_set,
+                                                                            dst_ports=dst_num_port_set,
+                                                                            protocols=protocols,
+                                                                            src_peers=src_res_pods,
+                                                                            dst_peers=dst_res_pods)
```

Comment:
here `dst_res_port` can be a `PortSet` with `excluded_named_ports` not empty. (for instance if the input policy had 'notPorts: a'

Reply:
Fixed (as explained above).